File: questão1/java/Carro.java
package questão1.java;

public class Carro {
    private String marca;
    private String modelo;
    private int ano;

    // Construtor da classe Carro
    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
    }

    // Método para exibir os detalhes do carro
    public String detalhes() {
        return "Carro: " + this.marca + " " + this.modelo + ", Ano: " + this.ano;
    }

    // Método principal
    public static void main(String[] args) {
        // Instanciando objetos da classe Carro
        Carro carro1 = new Carro("Toyota", "Corolla", 2020);
        Carro carro2 = new Carro("Honda", "Civic", 2019);
        Carro carro3 = new Carro("Ford", "Focus", 2018);

        // Exibindo os detalhes de cada carro
        System.out.println(carro1.detalhes());
        System.out.println(carro2.detalhes());
        System.out.println(carro3.detalhes());
    }
}

File: questão2/go/carro.go
package main

import "fmt"

// Definindo a struct Carro
type Carro struct {
    marca  string
    modelo string
    ano    int
}

// Método para exibir uma descrição detalhada do carro
func (c Carro) descricao() string {
    return fmt.Sprintf("Este é um %s %s de %d", c.marca, c.modelo, c.ano)
}

// Método para exibir detalhes básicos do carro
func (c Carro) detalhes() string {
    return fmt.Sprintf("Carro: %s %s, Ano: %d", c.marca, c.modelo, c.ano)
}

func main() {
    // Instanciando objetos da struct Carro
    carro1 := Carro{"Toyota", "Corolla", 2020}
    carro2 := Carro{"Honda", "Civic", 2019}

    // Exibindo a descrição detalhada de cada carro
    fmt.Println(carro1.descricao())
    fmt.Println(carro2.descricao())
}
